bash_docker/README.md

```diff
@@ -0,0 +1,145 @@
+# Bocker
+
+Docker implemented in around 100 lines of bash.
+
+-   [Prerequisites](#prerequisites)
+-   [Example Usage](#example-usage)
+-   [Functionality: Currently Implemented](#functionality-currently-implemented)
+-   [Functionality: Not Yet Implemented](#functionality-not-yet-implemented)
+-   [License](#license)
+
+## Prerequisites
+
+The following packages are needed to run bocker.
+
+-   btrfs-progs
+-   curl
+-   iproute2
+-   iptables
+-   libcgroup-tools
+-   util-linux >= 2.25.2
+-   coreutils >= 7.5
+
+Because most distributions do not ship a new enough version of util-linux you will probably need to grab the sources from [here](https://www.kernel.org/pub/linux/utils/util-linux/v2.25/) and compile it yourself.
+
+Additionally your system will need to be configured with the following :-
+
+-   A btrfs filesystem mounted under `/var/bocker`
+-   A network bridge called `bridge0` and an IP of 10.0.0.1/24
+-   IP forwarding enabled in `/proc/sys/net/ipv4/ip_forward`
+-   A firewall routing traffic from `bridge0` to a physical interface.
+
+For ease of use a Vagrantfile is included which will build the needed environment.
+
+Even if you meet the above prerequisites you probably still want to **run bocker in a virtual machine**. Bocker runs as root and among other things needs to make changes to your network interfaces, routing table, and firewall rules. **I can make no guarantees that it won't trash your system**.
+
+## Example Usage
+
+```
+$ bocker pull centos 7
```

Comment:
is it `docker`? If so, kindly review this file again and fix such `typos`

Reply:
bocker is the alias used to run the script as is evident from the directory   
`function bocker_init() { #HELP Create an image from a directory:\nBOCKER init <directory>
	uuid="img_$(shuf -i 42002-42254 -n 1)"
	if [[ -d "$1" ]]; then
		[[ "$(bocker_check "$uuid")" == 0 ]] && bocker_run "$@"
		btrfs subvolume create "$btrfs_path/$uuid" > /dev/null
		cp -rf --reflink=auto "$1"/* "$btrfs_path/$uuid" > /dev/null
		[[ ! -f "$btrfs_path/$uuid"/img.source ]] && echo "$1" > "$btrfs_path/$uuid"/img.source
		echo "Created: $uuid"
	else
		echo "No directory named '$1' exists"
	fi
}
`  
bocker was name given by wilcott and stands for BashDOCKER.

instead of screenshots, relevent conde snippets from the terminal have been included int he example usage in the readme file

## Example Usage

```
$ bocker pull centos 7
######################################################################## 100.0%
######################################################################## 100.0%
######################################################################## 100.0%
Created: img_42150

$ bocker images
IMAGE_ID        SOURCE
img_42150       centos:7

$ bocker run img_42150 cat /etc/centos-release
CentOS Linux release 7.1.1503 (Core)

$ bocker ps
CONTAINER_ID       COMMAND
ps_42045           cat /etc/centos-release

Reply:
please let me know if any other changes as and when required

Reply:
perfect!